dali/base/dasds.cpp

```diff
@@ -2482,6 +2493,27 @@ class CServerRemoteTree : public CRemoteTreeBase
         CHECKEDCRITICALBLOCK(SDSManager->treeRegCrit, fakeCritTimeout);
         SDSManager->queryAllNodes().addElem(this);
     }
+    void clear()
+    {
+        // always called inside SDSManager->treeRegCrit
+        if (serverId)
+        {
+            SDSManager->queryAllNodes().freeElem(serverId);
+            serverId = 0;
+        }
+    }
+    virtual bool removeTree(IPropertyTree *_child)
+    {
+        if (!_child)
+            return false;
+        Linked<CServerRemoteTree> child = QUERYINTERFACE(_child, CServerRemoteTree);
```

Comment:
e.g., a dbgassertex of the queryinterface, and a static cast for normal processing.